app/views/comments/_comments.html.erb

```diff
@@ -0,0 +1,9 @@
+<h3>Comments</h3>
+<% commentable.comments.each do |comment| %>
+  <div class="well">
+    <%= comment.body %>
+        <% if current_user == comment.user || current_user == article.user%>
```

Comment:
use `if policy(comment).destroy?` and that check the policy permission instead of duplicating the logic here.